src/rehydrator.ts

```diff
@@ -1,15 +1,24 @@
 import * as ReactDOM from "react-dom";
 
 import domElementToReact from "./dom-element-to-react";
+import ILoadedOptions from "./ILoadedOptions";
 import IOptions from "./IOptions";
 import IRehydrator from "./IRehydrator";
 
 const rehydratableToReactElement = async (
   el: Element,
   rehydrators: IRehydrator,
-  options: IOptions
+  options: ILoadedOptions
 ): Promise<React.ReactElement<any>> => {
-  const rehydratorName = el.getAttribute("data-rehydratable");
+  const rehydratorSelector = Object.keys(options.allSelectors).find(selector =>
```

Comment:
Worth considering switch to for loop:

![](https://nikitahl.com/images/find-array-items/find-array-item-results-snapshot.png)

[Source](https://nikitahl.com/how-to-find-an-item-in-a-javascript-array/). Need more data.